### PR TITLE
Use profile update

### PR DESCRIPTION
**Pull request by Bequiza**

### CODE REVIEW

File: app/src/main/java/se/rebeccazadig/bokholken/adverts/AdvertViewModel.kt [hunk 1]
```diff
@@ -9,7 +9,7 @@ import kotlinx.coroutines.launch
 import se.rebeccazadig.bokholken.data.Advert
 import se.rebeccazadig.bokholken.myAdverts.UiState
 
-class AdvertViewModel : ViewModel() {
+open class AdvertViewModel : ViewModel() {
```
Comment (ebesalti) [inline app/src/main/java/se/rebeccazadig/bokholken/adverts/AdvertViewModel.kt:12]: Why "open" here?

File: app/src/main/java/se/rebeccazadig/bokholken/myAdverts/MyAdvertsRepository.kt [hunk 2]
```diff
@@ -1,39 +1,23 @@
...
-    //     val bookListener = object : ValueEventListener {
-
-    //         override fun onDataChange(dataSnapshot: DataSnapshot) {
-    //             val fbfruits = mutableListOf<Advert>()
-    //             dataSnapshot.children.forEach { childsnap ->
-    //                 var tempad = childsnap.getValue<Advert>()!!
-    //                 tempad.adid = childsnap.key!!
-    //                 fbfruits.add(tempad)
-    //             }
+class MyAdvertsRepository private constructor() {
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/myAdverts/MyAdvertsRepository.kt:15]: Is this the only adverts-related repo? There should be only one.

File: app/src/main/java/se/rebeccazadig/bokholken/myAdverts/MyAdvertsRepository.kt [hunk 3]
```diff
@@ -1,39 +1,23 @@
...
 
-    //             aAdapter.filtreradeAnnonser = fbfruits
-    //             aAdapter.notifyDataSetChanged()
 
-    //             Log.i("pia11debug", fbfruits.toString())
-    //         }
+    private val myAuth = Firebase.auth
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/myAdverts/MyAdvertsRepository.kt:18]: Auth here?

File: app/src/main/java/se/rebeccazadig/bokholken/adverts/AdvertViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bequiza) [reply app/src/main/java/se/rebeccazadig/bokholken/adverts/AdvertViewModel.kt:12]: Forgot to remove. I tried something out and got the suggestion to make the class open. No intentions to extend.

File: app/src/main/java/se/rebeccazadig/bokholken/login/UserProfileFragment.kt [hunk 4]
```diff
@@ -32,19 +38,102 @@ class UserProfileFragment : Fragment() {
...
-            binding.contactTV.text = user.contact
-            binding.cityTV.text = user.city
+            binding.userNameTV.text = getString(R.string.user_details_username, user.name)
+            binding.contactTV.text = getString(R.string.user_details_contact, user.contact)
         }
 
         binding.editUserButton.setOnClickListener {
-            val editUserButton = UserProfileFragmentDirections.actionUserProfileFragment2ToUserInfoFragment()
-            findNavController().navigate(editUserButton)
+            val editButton = UserProfileFragmentDirections.actionUserProfileFragment2ToUserInfoFragment()
```
[the remark this answers is not in the document]
Comment (Bequiza) [reply app/src/main/java/se/rebeccazadig/bokholken/login/UserProfileFragment.kt:46]: fixed

File: app/src/main/java/se/rebeccazadig/bokholken/login/UserRepository.kt [hunk 5]
```diff
@@ -26,23 +26,6 @@ class UserRepository private constructor() {
...
-    //                 Pair(Result.Success, user)
-    //             } else {
-    //                 Pair(Result.Failure("User not found"), null)
-    //             }
-    //         } catch (e: Exception) {
-    //             Pair(Result.Failure(e.message ?: "Error fetching user!"), null)
-    //         }
-    //     }
-    // }
     suspend fun fetchUser(userId: String): Pair<Result, User?> {
```
[the remark this answers is not in the document]
Comment (Bequiza) [reply app/src/main/java/se/rebeccazadig/bokholken/login/UserRepository.kt:29]: fixed. Thanks for helping

**Pull request closed without merging**